packages/passport-interface/README.md

```diff
@@ -1,3 +1,167 @@
+
+For more detailed information about the package's functions and types, please refer to the [documentation site](https://docs.pcd.team/modules/_pcd_passport_interface.html).
+
+### Using the passport popup
+
+You can use the `usePassportPopupSetup` React hook to set up necessary passport popup logic on a specific route (e.g. `/popup`) and communicate with the passport webapp.
+
+```typescript
+import { usePassportPopupSetup } from "@pcd/passport-interface"
+
+export default function Popup() {
+    const error = usePassportPopupSetup()
+
+    return <div>{error}</div>
+}
+```
+
+Then the `usePassportPopupMessages` React hook can be used to process the results of your requests.
+
+```typescript
+import { usePassportPopupMessages } from "@pcd/passport-interface"
+
+export default function App() {
+    const [passportPCDString] = usePassportPopupMessages()
+
+     useEffect(() => {
+        console.log(passportPCDString)
+    }, [passportPCDString])
+}
+```
+
+### Adding a PCD to the passport
+
+You can generate a PCD and add it to the passport.
+
+```typescript
+import { EdDSAPCDPackage } from "@pcd/eddsa-pcd"
+import { constructPassportPcdProveAndAddRequestUrl, openPassportPopup } from "@pcd/passport-interface"
+import { ArgumentTypeName } from "@pcd/pcd-types"
+
+const url = constructPassportPcdProveAndAddRequestUrl<typeof EdDSAPCDPackage>(
+    process.env.PCDPASS_URL, // e.g. https://zupass.org.
+    window.location.origin + "/popup",
+    EdDSAPCDPackage.name,
+    {
+        id: {
+            argumentType: ArgumentTypeName.String
+    },
+        message: {
+            argumentType: ArgumentTypeName.StringArray,
+            value: ["0x342"]
+        },
+        privateKey: {
+            argumentType: ArgumentTypeName.String,
+            value: "0x42"
+        }
+    },
+    { title: "EdDSA Signature Proof" }
+)
+
+openPassportPopup("/popup", url)
+```
+
+Or add a previously generated serialized PCD.
+
+```typescript
+import { constructPassportPcdAddRequestUrl, openPassportPopup } from "@pcd/passport-interface"
+
+const url = constructPassportPcdAddRequestUrl(
+    process.env.PCDPASS_URL, // e.g. https://zupass.org.
+    window.location.origin + "/popup",
+    serializedPCD
+)
+
+openPassportPopup("/popup", url)
+```
+
+### Getting a PCD from the passport
+
+You can get a PCD from the passport without any proof.
+
+```typescript
+import { EdDSAPCDPackage } from "@pcd/eddsa-pcd"
+import { getWithoutProvingUrl, openPassportPopup } from "@pcd/passport-interface"
+
+const url = getWithoutProvingUrl(
+  process.env.PCDPASS_URL, // e.g. https://zupass.org.
+  window.location.origin + "/popup", // The popup route.
+  EdDSAPCDPackage.name // Name of PCD type.
+)
+
+openPassportPopup("/popup", url)
+```
+
+Or proving it.
```

Comment:
A little more verbose here would be nice

Reply:
Could you give me a specific example of using `constructPassportPcdGetRequestUrl`? Not sure if I've fully got it.